unsloth/models/llama.py

```diff
@@ -974,6 +986,9 @@ def _CausalLM_fast_forward(
         loss = None
         logit_softcapping = getattr(self.config, "final_logit_softcapping", 0)
         logit_scaling     = getattr(self.config, "logit_scale", 0)
+        if self.config.model_type == "granite":
+            # granite uses logit_scaling as key and they divide by the scale unlike cohere
+            logit_scaling = 1 / getattr(self.config, "logits_scaling", 1)
```

Comment:
Oh interesting - can you confirm it's not Cohere type logit scaling thanks :)

Reply:
- granite uses logit_scaling as key and they divide by the scale unlike cohere
notice that for granite, logits_scale is 16 and for cohere it is 0.125 (aka 1/8) in their respective configs
- [granite](https://github.com/huggingface/transformers/blob/4d1d0f29a493098e6bc6b904b82e29cb331827f5/src/transformers/models/granite/modeling_granite.py#L1103)
- [cohere](https://github.com/huggingface/transformers/blob/4d1d0f29a493098e6bc6b904b82e29cb331827f5/src/transformers/models/cohere/modeling_cohere.py#L1176)